### PR TITLE
fix: Exclude passthrough args after -- from scan input directories

### PR DESCRIPTION
**Pull request by danskmt**
## Pull Request Submission Checklist

- [x] Follows [CONTRIBUTING](https://github.com/snyk/cli/blob/main/CONTRIBUTING.md) guidelines
- [x] Commit messages are [release-note ready](https://github.com/snyk/cli/blob/main/CONTRIBUTING.md#writing-commit-messages), emphasizing _what_ was changed, not _how_.
- [x] Includes detailed description of changes
- [x] Contains risk assessment (Low | Medium | High)
- [x] Highlights breaking API changes (if applicable)
- [x] Links to automated tests covering new functionality
- [x] Includes manual testing instructions (if necessary)
- [ ] Updates relevant GitBook documentation (PR link: ___)
- [ ] Includes product update to be announced in the next stable release notes

## What does this PR do?

When a positional path is passed to a command together with `--` passthrough
args (e.g. `snyk test . -- -s /path/to/settings.xml`), the CLI was storing the
*entire* positional slice — including the tokens after `--` — into
`INPUT_DIRECTORY`. Those passthrough tokens then leak into the scan target list.

For `snyk test` with Risk Score enabled this caused a silent data loss: the
leaked `-s` token was misread as a package name by the downstream flow router,
which force-routed the scan to the legacy engine that never returns Risk Score.
The same leak also affects other slice consumers (SBOM test, monitor, secrets
test).

This fix trims the post-`--` passthrough args before setting `INPUT_DIRECTORY`,
so only real positional paths are treated as scan targets — matching the
existing behavior of `DetermineInputDirectory`.

## Where should the reviewer start?

- `cliv2/pkg/core/main.go` – `updateConfigFromParameter`: only positional args
  before `--` are stored as input directories.
- `cliv2/pkg/core/main_test.go` – `Test_runMainWorkflow_doubleDashArgsDoNotLeakIntoInputDirectory`
  covers `.`, relative, and absolute paths.

## How should this be manually tested?

With an org that has Risk Score enabled
(`useExperimentalRiskScore` + `useExperimentalRiskScoreInCLI`):

1. `snyk test -d` (from a scannable project) → output contains `Risk Score:` lines.
2. `snyk test . -d -- -s <any-settings-file>` → output should now also contain
   `Risk Score:` lines (previously absent), and debug should show
   `Using legacy flow: false`.

## What's the product update that needs to be communicated to CLI users?

Fixes a bug where passing an explicit path together with `--` passthrough
arguments could silently drop Risk Score data from `snyk test` results.

[CLI-1634]: https://snyksec.atlassian.net/browse/CLI-1634

### CODE REVIEW

File: cliv2/pkg/core/main.go [hunk 1]
```diff
@@ -162,10 +162,30 @@ func updateConfigFromParameter(config configuration.Configuration, args []string
 	}
 	config.Set(configuration.UNKNOWN_ARGS, doubleDashArgs)
 
-	// only consider the first positional argument as input directory if it is not behind a double dash.
-	if len(args) > 0 && !utils.Contains(doubleDashArgs, args[0]) {
-		config.Set(configuration.INPUT_DIRECTORY, args)
-		config.Set(localworkflows.ConfigurationNewAuthenticationToken, args[0])
```
Comment (danskmt) [inline cliv2/pkg/core/main.go:168]: In the previous code, `len(args) > 0` is always true if `doubleDashArgs` is not empty. The condition evaluates to true if `doubleDashArgs` doesn't contain `args[0]`, attempting to check: "is the first argument a real positional path, or did it come from behind the double dash?" However, this implementation introduced two critical bugs:

- **Argument Leakage**: Since Cobra strips out the "--" string but appends everything after it directly into args, setting `INPUT_DIRECTORY` to the raw args slice accidentally leaks the underlying tool's passthrough flags (like "-s") into our directory configuration.

- **The Duplication Edge Case**: Using `utils.Contains` completely breaks if the same string is used on both sides of the double dash (for example: `snyk test some/path -- command some/path`). Because `some/path` exists in doubleDashArgs, the code falsely assumes the first `some/path` is a passthrough flag, skips the block, and leaves `INPUT_DIRECTORY` completely unset.

Switching to length-based slicing (`len(args) - len(doubleDashArgs)`) avoids inspecting string content entirely, cleanly solving both the leakage and duplication bugs.
Image

File: cliv2/pkg/core/main_test.go [hunk 2]
```diff
@@ -185,71 +185,104 @@ func Test_CreateCommandsForWorkflowWithSubcommands(t *testing.T) {
...
 			}
 
-			positionalArgs := []string{expectedInputDir}
-			positionalArgs = append(positionalArgs, expectedUnknownArgs...)
-
-			rawArgs := []string{"app", "command", "--sad", expectedInputDir}
-			if len(expectedUnknownArgs) > 0 {
-				rawArgs = append(rawArgs, "--")
-				rawArgs = append(rawArgs, expectedUnknownArgs...)
+			if len(tc.wantUnknownArgs) == 0 {
```
Comment (PeterSchafer) [inline cliv2/pkg/core/main_test.go:281]: nitpick: the Equal assert below should work in both cases

**Review by PeterSchafer — APPROVED** [review]
